exporter/opentelemetry-exporter-otlp-proto-common/src/opentelemetry/exporter/otlp/proto/common/_internal/__init__.py

```diff
@@ -66,7 +66,11 @@ def _encode_resource(resource: Resource) -> PB2Resource:
     return PB2Resource(attributes=_encode_attributes(resource.attributes))
 
 
-def _encode_value(value: Any) -> PB2AnyValue:
+def _encode_value(
+    value: Any, allow_null: bool = False
```

Comment:
is there a need for `allow_null = False`?  could we allow nulls for everything?

Reply:
`_encode_value` is used via `_encode_key_value` by `_encode_attributes` that is used also by the metrics and trace encoders and did not want to change their behavior.

Reply:
A couple of thoughts:
- I'd imagine complex attributes on logs/events would also be prone to the same problem. It'd be nice to allow log attributes to have null values. So I think we should allow log attributes to have null values inside complex payloads.
- [minor] Allowing null values would not be a breaking change (it was not allowed before so nobody could add a null attribute), so I still wonder if additional logic is necessary.

Reply:
What do you mean with complex attributes? The function is recursive so we are handling null values nested in arrays and maps.

Reply:
I mean not just the body, but also logrecord attributes

Reply:
I'm fine with whatever approach you want to take - consider this a non-blocking discussion and this PR brings tons of value as it is.

Reply:
https://github.com/open-telemetry/opentelemetry-python/issues/4405 follow up issue for Riccardo